integration-tests/test/test_heterogenous_validators.py

```diff
@@ -1,6 +1,97 @@
+from random import Random
+
+from docker.client import DockerClient
 import pytest
 
-@pytest.mark.skip(reason="Skipped because we don't support bonding/unbonding in the new PoS yet")
-# see previous implementation for reference
-def test_heterogenous_validators() -> None:
-    pass
+from rchain.crypto import PrivateKey
+from . import conftest
+from .common import (
+    CommandLineOptions,
+    NonZeroExitCodeError,
+)
+from .rnode import (
+    bootstrap_connected_peer,
+    extract_validator_stake_from_bonds_validator_str,
+    started_bootstrap,
+    docker_network
+)
+from .wait import (
+    wait_for_node_sees_block,
+    wait_for_peers_count_at_least,
+    wait_for_blocks_count_at_least,
+)
+
+from .test_wallets import get_vault_balance
+
+BOOTSTRAP_KEY = PrivateKey.generate()
+BONDED_VALIDATOR_KEY = PrivateKey.generate()
+JOINING_VALIDATOR_KEY = PrivateKey.generate()
+READONLY_PEER_KEY = PrivateKey.generate()
```

Comment:
It looks like this introduces a source of non-determinism in the test.  Non-determinism causes flaky tests.  We should avoid it.  Is it possible to seed the underlying random number generator?